test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs

```diff
@@ -8546,6 +8550,418 @@ public async Task RestoreNetCore_ProjectProvidedRuntimeIdentifierGraphChange_Doe
             }
         }
 
+        [Theory]
+        [InlineData(new string[] { "win7-x86" }, new string[] { "win-x64" })]
+        [InlineData(new string[] { "win7-x86", "win-x64" }, new string[] { "win-x64" })]
+        [InlineData(new string[] { "win7-x86" }, new string[] { "win7-x86", "win-x64" })]
+        public void RestoreNetCore_PackagesLockFile_WithProjectChangeRuntimeAndLockedMode_FailsRestore(string[] intitialRuntimes, string[] updatedRuntimes)
+        {
+            // A project with RestoreLockedMode should fail restore if the project's runtime is changed between restores.
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Set up solution, project, and packages
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+                var tfm = "net45";
+
+                var projectA = SimpleTestProjectContext.CreateNETCore(
+                   "a",
+                   pathContext.SolutionRoot,
+                   new NuGetFramework[]{ NuGetFramework.Parse(tfm) });
+
+                projectA.Properties.Add("RestorePackagesWithLockFile", "true");
+                projectA.Properties.Add("RestoreLockedMode", "true");
```

Comment:
I'd normally expect RestoreLockedMode true and a lock file not existing to be an error. 

I feel like it'd lead the customer to a false sense of security. 

@anangaur What do you think? 

Should we file a bug for this?

Reply:
I agree. If RestoreLockedMode is set and lock file doesn’t exist, it should error stating “cannot file lock file...”

Reply:
If the lock file does not exist a lock file will be created. Why this is not as expected?

Reply:
Because the customer is indicated they want to be in locked mode. 

We can't deliver on the promise of locked mode if there's no lock file.

Reply:
Added a new issue: https://github.com/NuGet/Home/issues/8338